src/System.Runtime/tests/System/Int32Tests.cs

```diff
@@ -205,100 +208,110 @@ public static IEnumerable<object[]> Parse_Valid_TestData()
-            yield return new object[] { "-1E2", NumberStyles.AllowExponent | NumberStyles.AllowLeadingSign, null, -100 };
+            yield return new object[] { "1E2", 0, 3, NumberStyles.AllowExponent, null, 100 };
+            yield return new object[] { "1E+2", 0, 4, NumberStyles.AllowExponent, null, 100 };
+            yield return new object[] { "1e2", 0, 3, NumberStyles.AllowExponent, null, 100 };
+            yield return new object[] { "1E0", 0, 3, NumberStyles.AllowExponent, null, 1 };
+            yield return new object[] { "(1E2)", 0, 5, NumberStyles.AllowExponent | NumberStyles.AllowParentheses, null, -100 };
+            yield return new object[] { "-1E2", 0, 4, NumberStyles.AllowExponent | NumberStyles.AllowLeadingSign, null, -100 };
 
             NumberFormatInfo negativeFormat = new NumberFormatInfo() { PositiveSign = "|" };
-            yield return new object[] { "1E|2", NumberStyles.AllowExponent, negativeFormat, 100 };
+            yield return new object[] { "1E|2", 0, 4, NumberStyles.AllowExponent, negativeFormat, 100 };
 
             // AllowParentheses
-            yield return new object[] { "123", NumberStyles.AllowParentheses, null, 123 };
-            yield return new object[] { "(123)", NumberStyles.AllowParentheses, null, -123 };
+            yield return new object[] { "123", 0, 3, NumberStyles.AllowParentheses, null, 123 };
+            yield return new object[] { "(123)", 0, 5, NumberStyles.AllowParentheses, null, -123 };
 
             // AllowDecimalPoint
             NumberFormatInfo decimalFormat = new NumberFormatInfo() { NumberDecimalSeparator = "|" };
-            yield return new object[] { "67|", NumberStyles.AllowDecimalPoint, decimalFormat, 67 };
+            yield return new object[] { "67|", 0, 3, NumberStyles.AllowDecimalPoint, decimalFormat, 67 };
 
             // NumberFormatInfo has a custom property with length > 1
             NumberFormatInfo integerCurrencyFormat = new NumberFormatInfo() { CurrencySymbol = "123" };
-            yield return new object[] { "123123", NumberStyles.AllowCurrencySymbol, integerCurrencyFormat, 123 };
+            yield return new object[] { "123123", 0, 6, NumberStyles.AllowCurrencySymbol, integerCurrencyFormat, 123 };
 
             NumberFormatInfo integerPositiveSignFormat = new NumberFormatInfo() { PositiveSign = "123" };
-            yield return new object[] { "123123", NumberStyles.AllowLeadingSign, integerPositiveSignFormat, 123 };
+            yield return new object[] { "123123", 0, 6, NumberStyles.AllowLeadingSign, integerPositiveSignFormat, 123 };
         }
 
         [Theory]
         [MemberData(nameof(Parse_Valid_TestData))]
-        public static void Parse(string value, NumberStyles style, IFormatProvider provider, int expected)
+        public static void Parse(string value, int offset, int count, NumberStyles style, IFormatProvider provider, int expected)
```

Comment:
In the original cases offset ==0 and count == value.Length. I think adding explicit offset and length to all the existing data makes it less clearer. Would it be better to retain the original test for such cases, and add the offset/count variation for the new cases?

Reply:
To do that, I could see either:
a) using some sentinel value for offset/count to tell the test "ignore this, use the default", or
b) duplicate the tests and add new member datas with just the new entries with offsets/counts

While not ideal, it seemed better to me to just augment the existing tests and test cases to support offset/count.

Reply:
Right I was suggesting (b). Up to you.

Reply:
It seems like most of the tests remain as `offset = 0`, `count = string.Length`. We added a few that are actually doing the slicing, to test the fix. I am leaning towards (a), for example if we pass in -1, -1. Otherwise, it becomes cumbersome to add new test cases, because we would have to count the characters of each string.

Reply:
I refactored the new test data to avoid changing the others.